app/controllers/v0/messages_controller.rb

```diff
@@ -67,8 +68,55 @@ def thread
 
     private
 
+    def get_folder(folder_id)
+      folders = client.get_folder(folder_id)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, folder_id unless folders.present?
+      folders.data.first
+    end
+
+    # If page is not supplied, we are bringing back all results in the set. In that case
+    # adjust the per_page count to the TBD maximum number of messages allowed per MHV call
+    def pagination_params
+      folder_id = params[:folder_id].try(:to_i)
+      folder = get_folder(folder_id)
+      page = params[:page].try(:to_i)
+
+      if page.present?
+        pages = page..page
+        per_page = [params[:per_page].try(:to_i) || DEFAULT_PER_PAGE, MAXIMUM_PER_PAGE].min
+      else
+        # We need to test concatenation of multiple message GETS.
+        # Use only maximum per page once we can create more messages in a folder.
+        per_page = params[:per_page].try(:to_i) ||  MAXIMUM_PER_PAGE
+        pages = 1..(folder.count.to_f / per_page).ceil
+      end
+
+      { folder_id: folder_id, pages: pages, per_page: per_page, count: folder.count }
+    end
+
     def message_params
-      params.permit(:id, :category, :body, :recipient_id, :subject, :format)
+      # ActionController::Parameters No Longer Inherits from HashWithIndifferentAccess
+      # Gem message api uses keyword arguments and will not work with HashWithIndifferentAccess according
+      # to longstanding bug. Allegedly was fixed in Ruby 2.2, but having same issue in Ruby 2.3
+      hash = params.permit(:id, :category, :body, :recipient_id, :subject).to_h
```

Comment:
can we just call Hash.symbolize_keys ?

I'm a little weary of doing this though and I think the gem needs to maybe just be changed to support either string based keys or not use the new ruby 2.2 syntax for initializing the class. The reason I say this is, because symbols have been notorious for resulting in DDOS attacks in Rails applications. The Garbage collector handles symbols a little differently. It's probably OK in this context, but it would be nice if we can just fix how the message class accepts params instead?